data/src/main/scala/ch.epfl.scala.index.data/download/PlayWsDownloader.scala

```diff
@@ -113,7 +115,11 @@ trait PlayWsDownloader {
     }
     try {
       val response =
-        Await.result(processDownloads.runWith(Sink.seq), Duration.Inf)
+        if (toDownload.size == 1) {
```

Comment:
Initially, I thought the issue was with the old code. After getting a 403 response (abuse rate limit error) and waiting for 60 seconds, the code would bottleneck and instead of each thread resuming at the same time only one thread would resume at a time while the others would keep waiting for a minute which resulted in the whole github download step taking longer. The new code I added got rid of the bottleneck so all the threads would resume at more or less the same time and it takes 3 - 7 minutes less time than the old code.

I also thought the issue may have been something github was doing. Perhaps after triggering a 403 response they would keep sending 403 responses back for awhile which would make the code look like there was a bottleneck. But when I hit the github API in another shell while the download step was in the bottleneck, I didn't get any 403 responses so this probably is not the reason why the bottleneck is occurring.

I ran the old code and new code again today and verified the bottleneck still happens for the old code.